src/main/java/com/inscopemetrics/mad/statistics/StatisticFactory.java

```diff
@@ -60,80 +59,87 @@ public Statistic getStatistic(final String name) {
+    private static final ConcurrentMap<String, Statistic> STATISTICS_BY_NAME_AND_ALIAS;
     private static final Logger LOGGER = LoggerFactory.getLogger(StatisticFactory.class);
 
     static {
         // NOTE: Do not put log messages in static blocks since they can lock the logger thread!
-        final Map<String, Statistic> statisticByNameAndAlias = Maps.newHashMap();
-        final Set<Statistic> allStatistics = Sets.newHashSet();
-        final Set<Class<? extends Statistic>> statisticClasses = INTERFACE_DATABASE.findClassesWithInterface(Statistic.class);
-        for (final Class<? extends Statistic> statisticClass : statisticClasses) {
-            if (!statisticClass.isInterface() && !Modifier.isAbstract(statisticClass.getModifiers())) {
-                try {
-                    final Constructor<? extends Statistic> constructor = statisticClass.getDeclaredConstructor();
-                    if (!constructor.isAccessible()) {
-                        constructor.setAccessible(true);
-                    }
-                    final Statistic statistic = constructor.newInstance();
-                    allStatistics.add(statistic);
-                    checkedPut(statisticByNameAndAlias, statistic, statistic.getName());
-                    for (final String alias : statistic.getAliases()) {
-                        checkedPut(statisticByNameAndAlias, statistic, alias);
+        final ConcurrentMap<String, Statistic> statisticByNameAndAlias = Maps.newConcurrentMap();
+        try {
+            final ImmutableSet<ClassPath.ClassInfo> statisticClasses = ClassPath.from(StatisticFactory.class.getClassLoader())
+                    .getTopLevelClasses("com.inscopemetrics.mad.statistics");
+            for (final ClassPath.ClassInfo statisticClassInfo : statisticClasses) {
+                final Class<?> statisticClass = statisticClassInfo.load();
+                if (!statisticClass.isInterface() && !Modifier.isAbstract(statisticClass.getModifiers())
+                        && Statistic.class.isAssignableFrom(statisticClass)) {
+                    try {
+                        // The constructor type is implied by the assignability
+                        // of the statisticClass to the Statistic interface
+                        @SuppressWarnings("unchecked")
+                        final Constructor<? extends Statistic> constructor =
+                                (Constructor<? extends Statistic>) statisticClass.getDeclaredConstructor();
+                        if (!constructor.isAccessible()) {
+                            constructor.setAccessible(true);
+                        }
+                        checkedPut(statisticByNameAndAlias, constructor.newInstance());
+                    } catch (final InvocationTargetException | NoSuchMethodException
+                            | InstantiationException | IllegalAccessException e) {
+                        LOGGER.warn()
```

Comment:
This, combined with the "don't put log messages in static blocks" makes me nervous.